docs/learn/prompts.md

```diff
@@ -10,14 +10,36 @@ Prompts are the foundation of effective communication with Large Language Models
 
 The primary means of writing prompts in Mirascope is through prompt templates, which are just formatted strings (as they should be) with a few additional conveniences. This allows you to define prompts such that they are dynamic and reusable.
 
-For the purposes of explaining how prompt templates work, we will use:
+Prompt templates in Mirascope work similarly to f-strings in Python, but with added functionality specifically designed for LLM interactions. The `@prompt_template` decorator automatically injects variables from your class or function into the template.
+
+Let's look at a basic example:
+
+```python hl_lines="1 5"
+@prompt_template("Recommend a {genre} book")
+class BookRecommendationPrompt(BasePrompt):
+    genre: str
+
+prompt = BookRecommendationPrompt(genre="fantasy")
+print(prompt)
+# Output: Recommend a fantasy book
+```
+
+In this example:
+1. The `@prompt_template` decorator defines the template string.
+2. The `{genre}` placeholder in the template corresponds to the `genre` attribute of the class.
+3. When you create an instance of `BookRecommendationPrompt` with `genre="fantasy"`, the `genre` value is automatically injected into the template.
+
+This approach allows for flexible and reusable prompt creation, making it easy to generate variations of prompts by changing the input parameters.
+
+
+For more complex use cases, we'll use:
 
 - `@prompt_template`: a decorator for attaching a prompt template to a function or class
 - `BasePrompt`: a base class for writing provider-agnostic prompts.
 
-We will cover `BasePrompt` in more detail later. For now, all you need to know is that we inject fields with names that match template variables and that we've implemented the `__str__` method for easily viewing how template variables get injected:
+We'll cover `BasePrompt` in more detail later. For now, let's focus on how template variables are injected and how to view the resulting prompt:
```

Comment:
in line with a comment i make later, I think essentially we should show `BasePrompt` first briefly and then use the functional approach to discuss all of the various prompt templating features excluding the more complex ones (like computed fields).

we should mention here that we will cover `BasePrompt` in more detail in each of these respective sections.

key examples include talking about `prompt.run` in the Calls documentation or `computed_field` in the dynamic configuration section. i'm realizing it's weird to have these things separated and discussed multiple times.